From 03d5eff1f626309428d96b4d24505c8a833225e1 Mon Sep 17 00:00:00 2001
From: Ganesh Biradar <gbiradar@intrinsyc.com>
Date: Wed, 29 Aug 2018 15:15:11 +0530
Subject: [PATCH 282/334] [6558] ASoC: topology: Modify clock gating parameter
 parsing to switch

This improves the coding style of this piece of code

https://github.com/torvalds/linux/commit/fbeabd09236664f34ea3e4a9f7dcf5a0cdb7fc47#diff-eaf90d1933d444086df89343107f2579

Change-Id: I72762b5e62a716ab7f0d7268e7946967087aef9e
Signed-off-by: Ganesh Biradar <gbiradar@intrinsyc.com>
---
 sound/soc/soc-topology.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/sound/soc/soc-topology.c b/sound/soc/soc-topology.c
index 619c51b02fa6..78f1ab70bc01 100644
--- a/sound/soc/soc-topology.c
+++ b/sound/soc/soc-topology.c
@@ -1957,6 +1957,21 @@ static void set_link_hw_format(struct snd_soc_dai_link *link,
 
 		link->dai_fmt = hw_config->fmt & SND_SOC_DAIFMT_FORMAT_MASK;
 
+		/* clock gating */
+		switch (hw_config->clock_gated) {
+		case SND_SOC_TPLG_DAI_CLK_GATE_GATED:
+			link->dai_fmt |= SND_SOC_DAIFMT_GATED;
+			break;
+
+		case SND_SOC_TPLG_DAI_CLK_GATE_CONT:
+			link->dai_fmt |= SND_SOC_DAIFMT_CONT;
+			break;
+
+		default:
+			/* ignore the value */
+			break;
+		}
+
 		/* clock signal polarity */
 		invert_bclk = hw_config->invert_bclk;
 		invert_fsync = hw_config->invert_fsync;
-- 
2.24.1

